dh_segment/network/pretrained_models.py

```diff
@@ -10,114 +14,162 @@ def mean_substraction(input_tensor, means=_VGG_MEANS):
-                'resnet_v1_50/block2/unit_4/bottleneck_v1',
-                'resnet_v1_50/block3/unit_6/bottleneck_v1',
-                'resnet_v1_50/block4/unit_3/bottleneck_v1'
-            ]
-        else:
-            blocks_list = [
-                nets.resnet_v1.resnet_v1_block('block1', base_depth=64, num_units=3, stride=2),
-                nets.resnet_v1.resnet_v1_block('block2', base_depth=128, num_units=4, stride=2),
-                nets.resnet_v1.resnet_v1_block('block3', base_depth=256, num_units=6, stride=2),
-                nets.resnet_v1.resnet_v1_block('block4', base_depth=512, num_units=3, stride=1),
-            ]
-            desired_endpoints = [
-                'resnet_v1_50/conv1',
-                'resnet_v1_50/block1/unit_2/bottleneck_v1',
-                'resnet_v1_50/block2/unit_3/bottleneck_v1',
-                'resnet_v1_50/block3/unit_5/bottleneck_v1',
-                'resnet_v1_50/block4/unit_3/bottleneck_v1'
-            ]
-
-        net, endpoints = nets.resnet_v1.resnet_v1(input_tensor,
-                                                  blocks=blocks_list[:blocks],
-                                                  num_classes=None,
-                                                  is_training=is_training,
-                                                  global_pool=False,
-                                                  output_stride=None,
-                                                  include_root_block=True,
-                                                  reuse=None,
-                                                  scope='resnet_v1_50')
-
-        intermediate_layers = list()
-        for d in desired_endpoints[:blocks + 1]:
-            intermediate_layers.append(endpoints[d])
-
-        return net, intermediate_layers
+                    net = layers.max_pool2d(net, [2, 2], scope='pool1')
+                    if self.blocks >= 2:
+                        net = layers.repeat(net, 2, layers.conv2d, 128, [3, 3], scope='conv2')
+                        intermediate_levels.append(net)
+                        net = layers.max_pool2d(net, [2, 2], scope='pool2')
+                    if self.blocks >= 3:
+                        net = layers.repeat(net, 3, layers.conv2d, 256, [3, 3], scope='conv3')
+                        intermediate_levels.append(net)
+                        net = layers.max_pool2d(net, [2, 2], scope='pool3')
+                    if self.blocks >= 4:
+                        net = layers.repeat(net, 3, layers.conv2d, 512, [3, 3], scope='conv4')
+                        intermediate_levels.append(net)
+                        net = layers.max_pool2d(net, [2, 2], scope='pool4')
+                    if self.blocks >= 5:
+                        net = layers.repeat(net, 3, layers.conv2d, 512, [3, 3], scope='conv5')
+                        intermediate_levels.append(net)
+                        net = layers.max_pool2d(net, [2, 2], scope='pool5')
```

Comment:
The output of the last `max_pool2d` is not returned, shouldn't it be ?

Reply:
mmhh... good catch. In a way, it depends what we want to do with the model definition, it can make sense to keep the feature maps with the best resolution but the lower receptive field. But I think if we want to be consistent with what we had, `outputs` should have the pooled versions, thoughts?